### ISSUE TITLE
Migration away from Travis CI

### ISSUE THREAD

**Issue by larsrh** [typelevel/general#106]
tl;dr Travis CI has drastically reduced their OSS offerings.

The following repositories need to be migrated to GitHub Actions:

* [x] https://github.com/typelevel/cats-testkit-scalatest (typelevel/cats-testkit-scalatest#91)
* [x] https://github.com/typelevel/algebra (typelevel/algebra#238)
* [x] https://github.com/typelevel/kittens (typelevel/kittens#262)
* [x] https://github.com/typelevel/simulacrum
* [x] https://github.com/typelevel/discipline-scalatest (typelevel/discipline-scalatest#132)
* [x] https://github.com/typelevel/cats-tagless (typelevel/cats-tagless#169)
* [x] https://github.com/typelevel/frameless (typelevel/frameless#462)
* [x] https://github.com/typelevel/paiges (typelevel/paiges#304)
* [x] https://github.com/typelevel/spire
* [x] https://github.com/typelevel/cats-collections (typelevel/cats-collections#346)
* [x] https://github.com/typelevel/sbt-catalysts
* [x] https://github.com/typelevel/unique (typelevel/unique#162)
* [x] https://github.com/typelevel/semigroups (typelevel/semigroups#142)
* [x] https://github.com/typelevel/typelevel.github.com

The following steps need to be taken:

1. Create an issue in the project tracker (if not already there) and link it here
1. Delete `.travis.yml` (and `.mergify.yml`, if present)
1. Create the GitHub Actions workflow; use inspiration from other projects
1. If the project uses some form of deployment:
   * deployment to GitHub pages can be done using the regular `GITHUB_TOKEN`
   * deployment to Sonatype → ask a maintainer
1. Open a PR and alert me and @djspiewak 

We have until end of 2020 to complete this. Let's get it done.

**Comment by vic** [comment]
i guess I can help with this, will start migrating some repos over my free time this week.

**Comment by fthomas** [comment]
> Delete `.travis.yml` (and `.mergify.yml`, if present)

What's the reason for deleting `.mergify.yml` files? Mergify also works with GH Actions, [here](https://github.com/scala-steward-org/scala-steward/blob/24a3882f4c053143d28fb39e509e1152debd3565/.mergify.yml) is one example.

**Comment by larsrh** [comment]
It's true, but I see the risk that the `status-success` fields need to be kept updated manually. I'd rather disable auto-merging than risking broken merges.

**Comment by yanns** [comment]
This guide might help: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-travis-ci-to-github-actions

**Comment by dimitarg** [comment]
Maintainers will likely be already aware of this - I think https://github.com/djspiewak/sbt-github-actions might be a good implementation option.

https://github.com/dimitarg/weaver-test-extra is one example of mine using this in the wild; granted, its workflow is super simple but posting anyways. Sorry for spam if not helpful.

**Comment by jatcwang** [comment]
I can do `kittens`.

**Comment by vincenzobaz** [comment]
the SBT documentation now provides some advice about Github Actions: https://www.scala-sbt.org/1.x/docs/GitHub-Actions-with-sbt.html

**Comment by larsrh** [comment]
Done :rocket:
